test/util_test.go

```diff
@@ -52,3 +54,35 @@ func newKine(ctx context.Context, tb testing.TB) (*clientv3.Client, server.Backe
 	}
 	return client, backend
 }
+
+func newKineWithEndpoint(ctx context.Context, tb testing.TB, endpointConfig endpoint.Config) (*clientv3.Client, server.Backend) {
+	logrus.SetLevel(logrus.ErrorLevel)
+
+	config, backend, err := endpoint.ListenAndReturnBackend(ctx, endpointConfig)
+	if err != nil {
+		panic(err)
+	}
+	tlsConfig, err := config.TLSConfig.ClientConfig()
+	if err != nil {
+		panic(err)
+	}
+	client, err := clientv3.New(clientv3.Config{
+		Endpoints:   []string{endpointConfig.Listener},
+		DialTimeout: 5 * time.Second,
+		TLS:         tlsConfig,
+	})
+	if err != nil {
+		panic(err)
+	}
+	return client, backend
+}
+
+func newDqliteApp(tb testing.TB) (*app.App, error) {
+	dir := tb.TempDir()
+	app, err := app.New(dir, app.WithAddress("127.0.0.1:9001"))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create dqlite app: %w", err)
+	}
+
+	return app, nil
+}
```

Comment:
Let's adjust this to use `newKine()` instead, as we discussed offline.

Reply:
Done